client/src/main/java/org/apache/celeborn/client/read/CelebornInputStream.java

```diff
@@ -417,14 +418,27 @@ private boolean isExcluded(PartitionLocation location) {
 
     private PartitionReader createReaderWithRetry(
         PartitionLocation location, PbStreamHandler pbStreamHandler) throws IOException {
+      return createReaderWithRetry(location, pbStreamHandler, Optional.empty());
+    }
+
+    private PartitionReader createReaderWithRetry(
+        PartitionLocation location,
+        PbStreamHandler pbStreamHandler,
+        Optional<PartitionReaderCheckpointMetadata> checkpointMetadata)
+        throws IOException {
       Exception lastException = null;
+      PartitionReader reader = null;
```

Comment:
Could move this definition into line 437? The catch blocker does not use this variable.

Reply:
Ack